modules/index/src/main/java/org/apache/ignite/internal/index/IndexNodeFinishedRwTransactionsChecker.java

```diff
@@ -148,17 +151,28 @@ public <T> T inUpdateRwTxCountLock(Supplier<T> supplier) {
     }
 
     @Override
-    public HybridTimestamp minimumBeginTime() {
+    public long minimumRequiredTime() {
         readWriteLock.writeLock().lock();
 
+        int minRequiredVer;
+
         try {
-            // TODO https://issues.apache.org/jira/browse/IGNITE-22975 Improve minimum begin time determination
-            return txCatalogVersionByBeginTxTs.keySet().stream()
-                    .min(HybridTimestamp::compareTo)
-                    .orElse(clock.now());
+            Entry<Integer, Long> entry = txCountByCatalogVersion.lastEntry();
```

Comment:
`txCountByCatalogVersion` is initialized as `new ConcurrentSkipListMap<>()` (mind the lack of external comparator), what means all entries will be sorted naturally in ascending order. Here you get _last_ entry, given the ascending order this results in the biggest version used so far. But this method must return minimal required time.

The fact that all tests have passed freaks me out tbh

Reply:
:woman_facepalming:  fixed

> The fact that all tests have passed freaks me out tbh

Me too. The problem with the test was that only single RW transaction started on each node.

Reply:
Added separate unit test.